utils/splunk/splunk_telemetry_base.py

```diff
@@ -71,7 +71,17 @@ def _dispatch_and_await_search(self, instance, saved_searches):
 
         for saved_search in saved_searches:
             try:
+                persist_status_key = instance.instance_config.base_url + saved_search.name
+                if self.status.data.get(persist_status_key) is not None:
+                    for (sid, saved_search) in self.status.data[persist_status_key]:
+                        instance.splunkHelper.finalize_sid(sid, saved_search)
+                    self.status.data[persist_status_key] = []
```

Comment:
can we extract the set data and call `persist` as `def update_status` and reuse it below as well ?